lib/msf/base/simple/module.rb

```diff
@@ -18,10 +18,14 @@ module Module
   def _import_extra_options(opts)
     # If options were supplied, import them into the payload's
     # datastore
-    if (opts['Options'])
-      self.datastore.import_options_from_hash(opts['Options'])
-    elsif (opts['OptionStr'])
-      self.datastore.import_options_from_s(opts['OptionStr'])
+    if (value = opts['Options'])
+      if value.is_a?(String)
```

Comment:
This allows persisted jobs to still be loaded correctly, i.e. the scenario of `Options` being a string instead of hash